plugins/performance-lab/includes/site-health/bfcache-compatibility-headers/helper.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Helper functions used for Cache-Control headers for bfcache compatibility site health check.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+// @codeCoverageIgnoreStart
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+// @codeCoverageIgnoreEnd
+
+/**
+ * Tests the Cache-Control headers for bfcache compatibility.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function perflab_bfcache_check_compatibility(): array {
+	$result = array(
+		'label'       => __( 'The Cache-Control response header for pages ensures fast back/forward navigation.', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => '<p>' . esc_html__( 'If the Cache-Control response header includes directives like no-store, no-cache, or max-age=0 then it can prevent instant back/forward navigations (using the browser bfcache). Your site is configured properly.', 'performance-lab' ) . '</p>',
+		'actions'     => '',
+		'test'        => 'perflab_cch_cache_control_header_check',
+	);
+
+	$response = wp_remote_get(
+		home_url( '/' ),
+		array(
+			'headers'   => array( 'Accept' => 'text/html' ),
+			'sslverify' => false,
+		)
+	);
+
+	if ( is_wp_error( $response ) ) {
+		$result['label']       = __( 'Error checking cache settings', 'performance-lab' );
+		$result['status']      = 'recommended';
+		$result['description'] = '<p>' . wp_kses(
+			sprintf(
+				/* translators: %s is the error code */
+				__( 'There was an error while checking your Cache-Control response header. Error code: <code>%s</code> and the following error message:', 'performance-lab' ),
+				esc_html( (string) $response->get_error_code() )
+			),
+			array( 'code' => array() )
+		) . '</p><blockquote>' . esc_html( $response->get_error_message() ) . '</blockquote>';
```

Comment:
Why do we need the `blockquote` markup here? This feels odd for an error message. It would also be better for localization to use interpolation via `sprintf` and put a ` %s` after the `error message:` bit of the i18n message.

Reply:
The prior art for this comes from the Site Health test for Optimization Detective:

https://github.com/WordPress/performance/blob/762e52ad5998b1e064472c76875ddf44c243caaa/plugins/optimization-detective/site-health.php#L117-L124

https://github.com/WordPress/performance/blob/762e52ad5998b1e064472c76875ddf44c243caaa/plugins/optimization-detective/site-health.php#L145-L157

Reply:
I'm not sure what you mean about being better for localization? The markup strings here aren't inside of translation functions.

Reply:
I'm not sure how that would work with the existing HTML markup. Could you amend this PR with what you have in mind?

Reply:
Well, I guess it _does_ tie in with the markup in the sense that removing it would allow us to use `sprintf()` in the way I described above, which is better for localization. So I mentioned it as another reason to get rid of the unnecessary `blockquote` markup and instead simply display a single string that includes the actual error message.

Reply:
How is it better for localization? Core doesn't do this in many cases: 

https://github.com/search?q=repo%3AWordPress%2Fwordpress-develop+%2F_%5B_ex%5D%5C%28+%27%5B%5E%27%5D%2B%3A%27%2F&type=code

Compared with:

https://github.com/search?q=repo%3AWordPress%2Fwordpress-develop+%2F_%5B_ex%5D%5C%28+%27%5B%5E%27%5D%2B%3A+%25%28%5Cd%2B%5C%24%29%3Fs%27%2F&type=code

Reply:
For a specific example from core: https://github.com/WordPress/wordpress-develop/blob/20071e0fefcbd4653aefcacb8f67bd46cb5d28d6/src/wp-includes/class-wpdb.php#L1826-L1852

Reply:
I mean we don't need to get hung up on that. My main feedback was that I find the `blockquote` markup unnecessary. An error message can simply accompany the text that it is prefixed with, no need for another element.

FWIW I don't think there's any situation in Core where a `blockquote` or other element is used for such a message.

Reply:
Removed in f82cca94600d293bee36acbdea44305b2ebeb20c:

![image](https://github.com/user-attachments/assets/ce75bc68-71bb-42f7-904f-b2a9e1707bab)

However, I think it's easier to read when in a `BLOCKQUOTE`.